File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l=0
        max_count=0
        zeroes_flipped=0

        for r in range(len(nums)):
            if nums[r]==0:
                zeroes_flipped+=1
            
            while zeroes_flipped>k:
                if nums[l]==0:
                    zeroes_flipped-=1
                l+=1
            max_count=max(max_count, r-l+1)

        return max_count

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l =0,zeroes = 0, len=0,maxlen=0;
        for(int r=0;r<nums.length;r++)
        {
            if(nums[r]==1)
                len++;
            else if(nums[r]==0 && zeroes<k)
                {
                    len++;
                    zeroes++;
                }
            else
             { 
                    maxlen= Math.max(len,maxlen);
                    zeroes++;
                    len++;
                    while(zeroes>k)
                    {
                        if(nums[l]==0)
                        {
                            zeroes--;
                        }
                        len--;
                        l++;
                    }

            }
        }
        maxlen= Math.max(len,maxlen);

        return maxlen;
    }
}